src/fuzzer/analysis/typescript/ProgramDef.ts

```diff
@@ -1031,51 +1031,60 @@ export class ProgramDef {
-            ret[node.id.name] = {
-              name: node.id.name,
-              module: this._module,
-              src: this._src.substring(node.range[0], node.range[1]),
-              startOffset: node.range[0],
-              endOffset: node.range[1],
-              isExported: parent
-                ? parent.type === AST_NODE_TYPES.ExportNamedDeclaration
-                : false,
-              args: node.params
-                .filter((arg) => arg.type === AST_NODE_TYPES.Identifier)
-                .map((arg) => this._getTypeRefFromAstNode(arg as Identifier)),
-            };
+          try {
+            if (
+              // Arrow Function Definition: const xyz = (): void => { ... }
+              node.type === AST_NODE_TYPES.VariableDeclarator &&
+              parent !== undefined &&
+              parent.type === AST_NODE_TYPES.VariableDeclaration &&
+              node.init &&
+              node.init.type === AST_NODE_TYPES.ArrowFunctionExpression &&
+              node.id.type === AST_NODE_TYPES.Identifier &&
+              !isBlockScoped(node)
+            ) {
+              ret[node.id.name] = {
+                name: node.id.name,
+                module: this._module,
+                src:
+                  parent.kind +
+                  " " +
+                  this._src.substring(node.range[0], node.range[1]),
+                startOffset: node.range[0],
+                endOffset: node.range[1],
+                isExported: parent.parent
+                  ? parent.parent.type === AST_NODE_TYPES.ExportNamedDeclaration
+                  : false,
+                args: node.init.params
+                  .filter((arg) => arg.type === AST_NODE_TYPES.Identifier)
+                  .map((arg) => this._getTypeRefFromAstNode(arg as Identifier)),
+              };
+            } else if (
+              // Standard Function Definition: function xyz(): void => { ... }
+              node.type === AST_NODE_TYPES.FunctionDeclaration &&
+              node.id !== null &&
+              !isBlockScoped(node)
+            ) {
+              ret[node.id.name] = {
+                name: node.id.name,
+                module: this._module,
+                src: this._src.substring(node.range[0], node.range[1]),
+                startOffset: node.range[0],
+                endOffset: node.range[1],
+                isExported: parent
+                  ? parent.type === AST_NODE_TYPES.ExportNamedDeclaration
+                  : false,
+                args: node.params
+                  .filter((arg) => arg.type === AST_NODE_TYPES.Identifier)
+                  .map((arg) => this._getTypeRefFromAstNode(arg as Identifier)),
+              };
+            }
+          } catch (e: any) {
+            console.debug(
+              `Error processing function '${this._src.substring(
+                node.range[0],
+                node.range[1]
+              )}' in module '${this._module}': ${e.message}`
+            );
           }
```

Comment:
This change is a great step that helps us limit the blast radius of unsupported functions!

Could this PR be modified slightly to also return unsupported functions to the caller such that they may be populated to `_unsupportedFunctions` along with the error reason/message? 

This would involve changing the interface to `_findFunctions` such that it would return both supported and unsupported functions as well as differentiate among them and provide a message for those that are unsupported. 

Eventually we want to provide feedback to the user via the UI about why certain functions are not supported. Your change plus this small tweak would be a great step that helps us along that journey!

Reply:
Back in the office & thinking your change is even better with this last commit---planning to merge this shortly...